python/TestHarness/RunParallel.py

```diff
@@ -58,10 +58,10 @@ def __init__(self, harness, max_processes=None, average_load=64.0):
         self.big_queue = deque()
 
         # Jobs that have been finished
-        self.finished_jobs = set()
+        self.finished_jobs_dict = {}
 
         # List of skipped jobs to resolve prereq issues for tests that never run
-        self.skipped_jobs = set()
+        self.skipped_jobs_dict = {}
```

Comment:
yeah good point. I am only using finished_jobs for much of what I am attempting to do.